### DISCUSSION TITLE
Pagination, SendInteractive

### DISCUSSION THREAD

**Discussion by zaits07** [scrapli/scrapligo#194] [Q&A] [answered]
Hello.
I'm using the generic driver to get the configuration of a device.
Configuration is large text, so every page --MORE-- prompt appearing.
The problem is that **device not support pagination disabling**, and it expect **space char** to continue.
If i understand properly, there is SendInteractive(), wich can handle intermediate prompt --MORE--.
But (if i understand properly) SendInteractive() send return char for intermediate prompt.
What can i do?
Appreciate any suggestions.

Ideally would be method SendInteractive("command", "intermediate prompt", "intermediate input", "final prompt")
And logic:
1 - send "command" to device
2 - read until "intermediate prompt" or "final prompt"
3 - if "intermediate prompt" received, remove it from output, send "intermediate input" to device, continue from step to 2
4 - if "final prompt" received - finish output
so it can be called SendInteractive("show conf all", "^\s*--MORE--\s*", " ", "device>")
Just kick it in the right direction please.
Thanks.

**Comment by carlmontanari** [comment, marked as answer]
👋 hey @zaits07 

```go
package main

import (
	"bytes"
	"fmt"
	"github.com/scrapli/scrapligo/driver/generic"
	"github.com/scrapli/scrapligo/driver/options"
	"time"
)

func main() {
	var channelLog bytes.Buffer

	d, err := generic.NewDriver(
		"1.2.3.4",
		options.WithTransportType("system"),
		options.WithAuthNoStrictKey(),
		options.WithSSHConfigFileSystem(),
		options.WithChannelLog(&channelLog),
	)
	if err != nil {
		panic(err)
	}

	err = d.Open()
	if err != nil {
		panic(err)
	}

	defer d.Close()

	r, err := d.SendWithCallbacks("show run", []*generic.Callback{
		{
			Callback: func(driver *generic.Driver, s string) error {
				return driver.Channel.Write([]byte(" "), false)
			},
			Contains:    "--More--",
			ResetOutput: true,
		},
		{
			Contains: "end\n\nMYPROMPT#",
			Complete: true,
		},
	}, 3*time.Second)

	if err != nil {
		panic(err)
	}

	fmt.Println(r.Result)
}

```

this works on a cisco switch I have handy but with the "generic" driver (so pagination is not disabled). the "complete" callback is a bit fidgety since if if you just log right in and then execute this without the newlines before the prompt the initial buffer will have the prompt in it and it will "catch" the complete callback and the whole callback thing ill be done. you can solve this by sending some command before this such that the initial login buffer has been drained. probably would be good to fix the initial buffer containing data after login too... (I know this got addressed sorta recently in scrapli python).

anyway. that should work for ya I think.
